watertap/unit_models/tests/test_electrodialysis_0D.py

```diff
@@ -260,6 +262,44 @@ def test_performance_contents(self, electrodialysis_cell1):
             perform_dict["vars"]["Current efficiency for deionzation"]
         ) == pytest.approx(0.714, rel=5e-3)
 
+    @pytest.mark.component
+    def test_costing(self, electrodialysis_cell1):
+        m = electrodialysis_cell1
+        blk = m.fs.unit
+
+        # NOTE: This should probably move into the build of the model
+        #       and not be here after everything else (see next test)
+        m.fs.costing = WaterTAPCosting()
+
+        # Set costing var
+        m.fs.costing.electrodialysis_total_membrane_cost.set_value(86)
+        m.fs.costing.factor_electrodialysis_membrane_housing_replacement.set_value(0.2)
+
+        m.fs.unit.costing = UnitModelCostingBlock(
+            default={
+                "flowsheet_costing_block": m.fs.costing,
+                "costing_method_arguments": {},
+            },
+        )
+        m.fs.costing.cost_process()
+
+        assert_units_consistent(m)
+
+        assert degrees_of_freedom(m) == 0
+
+        results = solver.solve(m, tee=True)
+        assert_optimal_termination(results)
+
+        assert pytest.approx(388.6800, rel=1e-3) == value(
+            m.fs.costing.total_capital_cost
+        )
+        assert pytest.approx(45.86804, rel=1e-3) == value(
+            m.fs.costing.total_operating_cost
+        )
+        assert pytest.approx(777.3600, rel=1e-3) == value(
+            m.fs.costing.total_investment_cost
+        )
+
```

Comment:
Are there any references these numbers can be compared against to make sense?

Reply:
We would need to look in the literature. Presumably, if the voltage and power usage is correct, then this should be accurate.